pkg/ingresscache/safetrie.go

```diff
@@ -146,25 +152,77 @@ func (st *SafeTrie) IsEmpty() bool {
 	return walkResult == nil
 }
 
-// TODO - will be removed once moving into efficient pathFunctionNames implementation (i.e. not using slices)
-func excludeElemFromSlice(slice []string, elem string) []string {
-	// Assuming len(slice) <= 2 based on the ingress validations of: https://github.com/nuclio/nuclio/pkg/platform/kube/platform.go
-	switch len(slice) {
-	case 1:
-		if slice[0] == elem {
-			return []string{}
-		}
-		return slice
-	case 2:
-		if slice[0] == elem {
-			return []string{slice[1]}
-		}
-		if slice[1] == elem {
-			return []string{slice[0]}
-		}
-		// elem not found, return original slice
-		return slice
-	default:
-		return slice
+// ----- implementations for FunctionTarget interface -----
+
+type SingleTarget struct {
+	functionName string
+}
```

Comment:
This can also be a wrapper for a string
```suggestion
type SingleTarget string
```

If we're talking about optimization, what is the difference between the two?